src/goto-instrument/contracts/utils.cpp

```diff
@@ -242,3 +242,30 @@ bool is_assigns_clause_replacement_tracking_comment(const irep_idt &comment)
   return id2string(comment).find(ASSIGNS_CLAUSE_REPLACEMENT_TRACKING) !=
          std::string::npos;
 }
+
+assignst widen_assigns(const assignst &assigns)
+{
+  assignst result;
+
+  havoc_utils_is_constantt is_constant(assigns);
+
+  for(auto it = assigns.begin(); it != assigns.end(); ++it)
+  {
+    if(it->id() == ID_index || it->id() == ID_dereference)
+    {
+      address_of_exprt address_of_expr(*it);
+
+      // index or offset is non-constant.
+      if(!is_constant(address_of_expr))
+      {
+        result.emplace(
+          pointer_object_exprt(address_of_expr, address_of_expr.type()));
```

Comment:
Please use `pointer_object(address_of_expr)` to ensure the right type (the type expected by `pointer_object_exprt` is a numerical one, not a pointer type).